Just gave this another try. When I build with `+doc`, I get errors like this and the build fails:
```
make[4]: .../opt/spack/linux-ubuntu18.04-x86_64/gcc-10.2.0/qt-5.15.2-aawjt6mlcbihiljmczi6jtis5sgsvfy6/bin/qtattributionsscanner: Command not found
Makefile.qmake-aux:233: recipe for target 'qtattributionsscanner' failed
make[4]: *** [qtattributionsscanner] Error 127
```
Any idea?

--- a/var/spack/repos/builtin/packages/qt/package.py
+++ b/var/spack/repos/builtin/packages/qt/package.py
@@ -584,10 +584,14 @@ class Qt(Package):
 
     def build(self, spec, prefix):
         make()
-        if '+doc' in spec:
-            make('docs')
 
     def install(self, spec, prefix):
         make("install")
-        if '+doc' in spec:
+
+    # Documentation generation requires the doc tools to be installed.
+    # @when @run_after currently seems to ignore the 'when' restriction.
+    @run_after('install')
+    def install_docs(self):
+        if '+doc' in self.spec:
+            make('docs')
             make('install_docs')

